File: src/o2a/providers/instrument/calibration.zig
const calibration = @import("../../../kernels/spectra/calibration.zig");
const Scene = @import("../../../model/Scene.zig").Scene;
const SpectralChannel = @import("../../../model/Instrument.zig").SpectralChannel;

pub const Calibration = calibration.Calibration;

pub fn calibrationForScene(scene: *const Scene, channel: SpectralChannel) calibration.Calibration {
    const controls = scene.observation_model.resolvedChannelControls(channel);
    return .{
        .gain = controls.multiplicative_offset,
        .offset = controls.additive_offset,
        .wavelength_shift_nm = controls.wavelength_shift_nm,
        .stray_light = controls.stray_light,
    };
}

File: rust_src/forward_model/implementations/instrument/calibration.rs
use crate::{
    forward_model::instrument_grid::spectral_math::calibration::Calibration,
    input::{instrument::SpectralChannel, scene::Scene},
};

pub fn calibration_for_scene(scene: &Scene, channel: SpectralChannel) -> Calibration {
    let controls = scene.observation_model.resolved_channel_controls(channel);
    Calibration {
        gain: controls.multiplicative_offset,
        offset: controls.additive_offset,
        wavelength_shift_nm: controls.wavelength_shift_nm,
        stray_light: controls.stray_light,
    }
}
